internal/cmd/controller/gitops/reconciler/gitjob.go

```diff
@@ -48,18 +48,25 @@ const (
 
 func (r *GitJobReconciler) createJobAndResources(ctx context.Context, gitrepo *v1alpha1.GitRepo, logger logr.Logger) error {
 	logger.V(1).Info("Creating Git job resources")
+
 	if err := r.createJobRBAC(ctx, gitrepo); err != nil {
+		gitjobsCreatedFailure.Inc(gitrepo)
```

Comment:
Wouldn't be easier to record metrics in the caller instead? Same for the other case below.